crdb/tx.go

```diff
@@ -50,6 +50,17 @@ func ExecuteTx(db *sql.DB, fn func(*sql.Tx) error) (err error) {
 	if err != nil {
 		return err
 	}
+	return ExecuteBegunTx(tx, func() error { return fn(tx) })
+}
+
+type Tx interface {
+	Exec(query string, args ...interface{}) (sql.Result, error)
+	Commit() error
+	Rollback() error
+}
+
+// ExecuteBegunTx runs fn inside tx which should already have begun.
+func ExecuteBegunTx(tx Tx, fn func() error) (err error) {
```

Comment:
I'd like to find a better name than this. `Begun` is an odd term. Perhaps `ExecuteWithTx` or `ExecuteInTx`.